### PR TITLE
NLU-4012: [ar-ae] DD/MM dates resolving correctly in context

### PR DESCRIPTION
**Pull request by kevinwalshgen**
Currently 21/3 will resolve correctly on its own but not when referenced in a sentence.

### CODE REVIEW

File: Specs/DateTime/Arabic/DateParser.json [hunk 1]
```diff
@@ -1495,11 +1494,10 @@
     "Context": {
       "ReferenceDateTime": "2016-11-07T00:00:00"
     },
-    "IgnoreResolution": "true",
-    "NotSupportedByDesign": "java, javascript, python",
+    "NotSupportedByDesign": "java, javascript",
     "Results": [
       {
-        "Text": "إلى 27",
```
Comment (colm-dillon) [inline Specs/DateTime/Arabic/DateParser.json:1502]: Were these invalid test cases?

**Review by colm-dillon — APPROVED** [review]

File: Python/libraries/recognizers-date-time/recognizers_date_time/resources/arabic_date_time.py [hunk 2]
```diff
@@ -140,6 +140,7 @@ class ArabicDateTime:
     DateExtractor9L = f'\\b({WeekDayRegex}\\s+)?{DayRegex}\\s*/\\s*{MonthNumRegex}{DateExtractorYearTermRegex}(?![%])\\b'
     DateExtractor9S = f'\\b({WeekDayRegex}\\s+)?{DayRegex}\\s*[/-]\\s*{MonthNumRegex}(?![%])\\b'
     DateExtractorA = f'\\b({WeekDayRegex}\\s+)?{BaseDateTime.FourDigitYearRegex}\\s*[/\\\\\\-\\.]\\s*({MonthNumRegex}|{MonthRegex})\\s*[/\\\\\\-\\.]\\s*{DayRegex}'
+    DateExtractorB = f'(?:(?<=إلى|لل|يوم)\s*)({DayRegex})(?![\\./-]\d|\d)'
```
Comment (kevinwalshgen) [inline Python/libraries/recognizers-date-time/recognizers_date_time/resources/arabic_date_time.py]: Need the basic date capture to capture the "day/for/to 27" use cases but not "day/for/to 27/03" which will get picked up in the implicit date regexes. if not captured here 'day' will get picked up in the implicit regexes and today's date is resolved.
Hence added this extra case here.

File: Specs/DateTime/Arabic/DateExtractor.json [hunk 3]
```diff
@@ -796,9 +796,9 @@
     "NotSupportedByDesign": "java, javascript",
     "Results": [
       {
-        "Text": "إلى 27",
-        "Start": 4,
-        "Length": 6,
+        "Text": " 27",
```
Comment (kevinwalshgen) [inline Specs/DateTime/Arabic/DateExtractor.json:799]: So the extraction no longer returns 'day 27' as it just picks up ' 27' from the basic date regexes and this is passed to the parser which figures out its a XXXX-XX-27 format to resolve to the correct date.

**Review by andrew-gradinari — APPROVED** [review]

**Review by BobbyMartins5 — APPROVED** [review]

**Review by Conor-Keaney — APPROVED** [review]

File: Python/libraries/recognizers-date-time/recognizers_date_time/resources/arabic_date_time.py (hunk 2 above)
Comment (andrew-gradinari) [reply Python/libraries/recognizers-date-time/recognizers_date_time/resources/arabic_date_time.py]: Shouldn't we escape backslashes in the regex?

**Pull request merged**